### ISSUE TITLE
isBetween() produces false positive when this moment is in UTC and the fromValue and toValue were formatted string dates

### ISSUE THREAD

**Issue by duydnguyen07** [moment/moment#5754]
**Describe the bug**
When the currentMoment is in UTC and the startDate and endDate are passed in as string, the result of the isBetween function is expectedly wrong. 

**To Reproduce**
Steps to reproduce the behavior, try to execute the following. 
const currentMoment = moment().year(2020).month(9).date(14).tz('UTC');
currentMoment.isBetween('2020-10-15', '2020-10-16', 'date', '[]')

**Expected behavior**
The result of the above will yield true, while the expected result is false. In plain English, the currentMoment of the date 2020-10-14 is NOT in between '2020-10-15' and '2020-10-16', yet the function responded as true. 

**Desktop (please complete the following information):**
 - Browser: chrome
 - Version: 2.29.1

**Moment-specific environment**

**(new Date()).toString():** Wed Oct 14 2020 15:19:38 GMT+0200 (Central European Summer Time)
**toLocaleString():** 10/14/2020, 3:19:38 PM
**getTimezoneOffset():** -120
**userAgent:** Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.75 Safari/537.36
**version:** 2.29.1

**Comment by duydnguyen07** [comment]
I'm more than happy to contribute a fix if this is recognized as a real bug.

**Comment by halilhicyilmaz** [comment]
const timeSlot = moment(slot.format('HH:mm'))
const twoHoursFromNow = moment(moment.tz("Europe/Istanbul").add(2, "hours").format('HH:mm'))

const isBeforeTwoHoursFromNow = timeSlot.isBefore(twoHoursFromNow, "hours")

This is my workaround solution to bug. Format all the times then change them into moment again before compare.

**Comment by ichernev** [comment]
The first line uses UTC, for whatever reason, and the "time strings" don't have a timezone, so it's local time. Also the first instance has the time from your current time, and when you change the tz to UTC it can change the date. Just pass proper boundaries. There are moment constructors that can achieve it ergonomically.
